src/main/scala/tech/cryptonomic/conseil/Lorre.scala

```diff
@@ -469,15 +475,28 @@ object Lorre extends App with TezosErrors with LazyLogging with LorreAppConfig w
           Option[Voting.BallotCounts],
           Option[Voting.BallotCounts]
       )
-    ] = blocks.map { block =>
+    ] = blocks.flatMap { block =>
       val proposal = proposals.get(block.data.hash).flatten
       val listing = listings.get(block.data.header.level).toList.flatten
       val prevListings = listings.get(block.data.header.level - 1).toList.flatten
       val listingByBlock = listing.diff(prevListings)
       val ballot = ballots.get(block).toList.flatten
       val ballotCountPerCycle = ballotCountsPerCycle.get(block)
       val ballotCountPerLevel = ballotCountsPerLevel.get(block)
-      (block.data, proposal, listing, listingByBlock, ballot, ballotCountPerCycle, ballotCountPerLevel)
+      val proposalHashesForBlock = proposalHashes.get(block).toList.flatten
+
+      List((block.data, proposal, listing, listingByBlock, ballot, ballotCountPerCycle, ballotCountPerLevel)) ++
```

Comment:
Instead of building a singleton List with a single value and concatenate that with the proposal's list data, It might be easier to prepend the fist element to the rest with the cons `::` operator?

`(block.data, proposal, ...) :: proposalHashesFroBlock.map {...}` ?